Core/include/Acts/EventData/TrackProxy.hpp

```diff
@@ -687,14 +687,20 @@ class TrackProxy {
 
       // reverse using standard linked list reversal algorithm
       reverseTrackStates();
+    } else {
+      // just copy the track state indices
+      tipIndex() = other.tipIndex();
+      stemIndex() = other.stemIndex();
```

Comment:
@paulgessinger I put this in here just to have a brief discussion. To me this seemed like the expected behavior if somebody copies a track and just modifies the properties but not the states that also the tip would point to the same as the previous track. What do you think?

Just a slight preference from my side happy to revert

Reply:
One problem is that these are only valid if the track is copied within the same container though, right? If you do this generically, you might get corrupted tracks.

I see how this can be useful as a "shallow copy", however.

Reply:
Right I did not think of that case... that "shallow copy" would look like this `trackProxy.copy(withStates=false)`? So we can guarantee it lands in the same container?

That might be useful. For this PR I will just drop the change

Reply:
> Right I did not think of that case... that "shallow copy" would look like this `trackProxy.copy(withStates=false)`? So we can guarantee it lands in the same container?

I'd probably literally call it `trackProxy.copyShallow` and the only thing it does is copy without deep copying the track states. If you want a deep copy, you can still call the other function.

> That might be useful. For this PR I will just drop the change

Sounds good!